src/NexusMods.App.UI/Helpers/TreeDataGridHelpers.cs

```diff
@@ -0,0 +1,76 @@
+    /// <returns>A configured HierarchicalTreeDataGridSource.</returns>
+    /// <typeparam name="TNode">Type of node which derives from <see cref="TreeNodeVM{TItem,TKey}"/></typeparam>
+    /// <typeparam name="TItem">Type of item which is stored inside <typeparamref name="TNode"/>'s value</typeparam>
+    /// <typeparam name="TKey">Type of item which is stored inside <typeparamref name="TNode"/>'s key</typeparam>
+    public static HierarchicalTreeDataGridSource<TNode> CreateTreeSourceWithSingleCustomColumn<TNode, TItem, TKey>(
+        TNode treeRoot)
+        where TNode : TreeNodeVM<TItem, TKey>
+        where TItem : class, IViewModelInterface
+        where TKey : notnull
+    {
+        return CreateTreeSourceWithSingleCustomColumn<TNode, TItem, TKey>(Enumerable.Repeat(treeRoot, 1));
+    }
+
+    /// <summary>
+    /// Generates a HierarchicalTreeDataGridSource with a custom column for a given tree structure.
+    /// </summary>
+    /// <param name="treeRoots">An observable collection of the tree roots.</param>
+    /// <returns>A configured HierarchicalTreeDataGridSource.</returns>
+    /// <typeparam name="TNode">Type of node which derives from <see cref="TreeNodeVM{TItem,TKey}"/></typeparam>
+    /// <typeparam name="TItem">Type of item which is stored inside <typeparamref name="TNode"/>'s value</typeparam>
+    /// <typeparam name="TKey">Type of item which is stored inside <typeparamref name="TNode"/>'s key</typeparam>
+    public static HierarchicalTreeDataGridSource<TNode> CreateTreeSourceWithSingleCustomColumn<TNode, TItem, TKey>(
+        IEnumerable<TNode> treeRoots) 
+        where TNode : TreeNodeVM<TItem, TKey>
+        where TItem : class, IViewModelInterface
+        where TKey : notnull
+    {
+        var locator = Locator.Current.GetService<IViewLocator>();
+        return new HierarchicalTreeDataGridSource<TNode>(treeRoots)
+        {
+            Columns =
+            {
+                new HierarchicalExpanderColumn<TNode>(
+                    new TemplateColumn<TNode>(null,
+                        new FuncDataTemplate<TNode>((node, _) =>
+                            {
+                                // This should never be null but can be during rapid resize, due to
+                                // virtualization shenanigans. Think this is a control bug, but well, gotta work with what we have.
+                                // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+                                if (node == null)
+                                    return new Control();
+                                    
+                                // Very sus but it works, t r u s t.
+                                var view = locator!.ResolveView(node.Item);
+                                var ctrl = view as Control;
+                                ctrl!.DataContext = node.Item;
+                                return ctrl;
```

Comment:
These should have some `Debug.Assert` to make sure the things are actually not-null.